sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -128,6 +129,38 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
         ShimLoader.setSparkShimProviderClass(conf.shimsProviderOverride.get)
       }
 
+      // Compare if the cudf version mentioned in the classpath is equal to the version which plugin
+      // expects. If there is a version mismatch, throw error. This check can be disabled by
+      // setting this config spark.rapids.cudf-version-override=true
+      if (!conf.cudfVersionOverride) {
+        val props = new Properties
+        val cudfClassLoader = classOf[ai.rapids.cudf.ColumnVector].getClassLoader
+        try {
+          props.load(cudfClassLoader.getResourceAsStream("cudf-java-version-info.properties"))
+        } catch {
+          case t: Throwable => {
+            logError("cudf properties file not found.", t)
+          }
+        }
+        val cudfVersion = props.get("version").toString
```

Comment:
This needs to check for the property being missing to avoid the NPE and instead throw a much more useful error message.  Otherwise we're going to have to dig up the source to figure out what the NPE is trying to tell us.

Reply:
Have added the check and it throws exception if property is not found. Please let me know if the message is okay or it needs to be verbose.